multicluster/cmd/multicluster-controller/clusterset_webhook.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2022 Antrea Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	mcv1alpha1 "antrea.io/antrea/multicluster/apis/multicluster/v1alpha1"
+)
+
+//+kubebuilder:webhook:path=/validate-multicluster-crd-antrea-io-v1alpha1-clusterset,mutating=false,failurePolicy=fail,sideEffects=None,groups=multicluster.crd.antrea.io,resources=clustersets,verbs=create;update,versions=v1alpha1,name=vclusterset.kb.io,admissionReviewVersions={v1,v1beta1}
+
+// ClusterSet validator
+type clusterSetValidator struct {
+	Client    client.Client
+	decoder   *admission.Decoder
+	namespace string
+}
+
+// Handle handles admission requests.
+func (v *clusterSetValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	clusterSet := &mcv1alpha1.ClusterSet{}
+	err := v.decoder.Decode(req, clusterSet)
+	if err != nil {
+		klog.ErrorS(err, "Error while decoding")
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	oldClusterSet := &mcv1alpha1.ClusterSet{}
+	err = v.Client.Get(ctx, types.NamespacedName{Namespace: v.namespace, Name: clusterSet.GetName()}, oldClusterSet)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return admission.Errored(http.StatusPreconditionFailed,
+			fmt.Errorf("failed to get existing ClusterSet %s", clusterSet.GetName()))
+	}
+
+	// Check if there is any existing ClusterSet.
+	clusterSetList := &mcv1alpha1.ClusterSetList{}
+	if err := v.Client.List(context.TODO(), clusterSetList, client.InNamespace(v.namespace)); err != nil {
+		klog.ErrorS(err, "Error reading ClusterSet", "namespace", v.namespace)
```

Comment:
Maybe "Namespace" is more consistent with other resources.

Reply:
Fixed